### ISSUE TITLE
Adding `@property` to already decorated method leads to errors

### ISSUE THREAD

**Issue by wpk-nist-gov** [microsoft/pyright#7667]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

If you have a question about typing or a behavior that you’re seeing in Pyright (as opposed to a bug report or enhancement request), consider posting to the [Pyright discussion forum](https://github.com/microsoft/pyright/discussions).

**Describe the bug**
Updating to pyright==1.1.358 breaks type hints for adding decorating properties on top of another decorator.

**Code or Screenshots**
The below example works fine with `pyright < 1.1.358`, but breaks for `pyright == 1.1.358`:

```python
from typing import ParamSpec, TypeVar, Callable

P = ParamSpec("P")
R = TypeVar("R")

def decorate(func: Callable[P, R]) -> Callable[P, R]:
    def wrapper(*args: P.args, **kwargs: P.kwargs) -> R:
        print("hello")
        return func(*args, **kwargs)
    return wrapper

class Example:
    @decorate
    def meth(self) -> int:
        return 1

    @property
    @decorate
    def prop(self) -> int:
        return 1

x = Example()
reveal_type(x.meth)
reveal_type(x.meth())
reveal_type(x.prop)
```

results in the output (for pyright==1.1.358):
```bash
/../tmp.py
  /../tmp.py:23:13 - information: Type of "x.meth" is "() -> int"
  /../tmp.py:24:13 - information: Type of "x.meth()" is "int"
  /../tmp.py:25:13 - information: Type of "x.prop" is "Unknown"
  /../tmp.py:25:15 - error: Cannot access member "prop" for type "Example"
    Failed to call method "__get__" for descriptor class "property" (reportAttributeAccessIssue)
```

and for pyright==1.1.357:
```bash
/../tmp.py
  /../tmp.py:23:13 - information: Type of "x.meth" is "() -> int"
  /../tmp.py:24:13 - information: Type of "x.meth()" is "int"
  /../tmp.py:25:13 - information: Type of "x.prop" is "int"
```

**VS Code extension or command-line**
Running from command line.

**Comment by erictraut** [comment]
Thanks for the bug report, and apologies for the regression. I've determined that this problem was introduced as part of [this commit](https://github.com/microsoft/pyright/commit/3b70417dd549f6663b8f86a76f75d8dfd450f4a8). I'll continue to investigate.

**Comment by erictraut** [comment]
This will be fixed in the next release.

**Comment by wpk-nist-gov** [comment]
I'm stillrunning into similar issues with `pyright==1.1.359`.

The below code works with `pyright<1.1.358`, but fails otherwise.

```python
from typing import ParamSpec, TypeVar, Callable, TypeAlias, Protocol, Concatenate

P = ParamSpec("P")
R = TypeVar("R")
S = TypeVar("S", bound="HasAttr")
Dec: TypeAlias = Callable[Concatenate[S, P], R]


class HasAttr(Protocol):
    my_attr: str

def decorate(func: Dec[S, P, R]) -> Dec[S, P, R]:
    def wrapper(self: S, /, *args: P.args, **kwargs: P.kwargs) -> R:
        print("with attr", self.my_attr)

        return func(self, *args, **kwargs)
    return wrapper


# This would fail because it's not an instance of HasAttr.  This is good.
# class FailOnPurpose:
#     @decorate
#     def meth(self) -> int:
#         return 1

class Works:
    my_attr: str = "Works"

    @decorate
    def meth(self) -> int:
        return 1


    # But this fails
    @property
    @decorate
    def prop(self) -> int:
        return 1


x = Works()
reveal_type(x.meth)
reveal_type(x.meth())
reveal_type(x.prop)
```

With `pyright==1.1.359`:
```bash
/tmp2.py
  /tmp2.py:46:13 - information: Type of "x.meth" is "() -> int"
  /tmp2.py:47:13 - information: Type of "x.meth()" is "int"
  /tmp2.py:49:13 - information: Type of "x.prop" is "Unknown"
  /tmp2.py:49:15 - error: Cannot access attribute "prop" for class "Works"
    Failed to call method "__get__" for descriptor class "property" (reportAttributeAccessIssue)
1 error, 0 warnings, 3 informations
```

With `pyright==1.1.357`:
```bash
/tmp2.py
  /tmp2.py:46:13 - information: Type of "x.meth" is "() -> int"
  /tmp2.py:47:13 - information: Type of "x.meth()" is "int"
  /tmp2.py:49:13 - information: Type of "x.prop" is "int"
0 errors, 0 warnings, 3 informations
```

I use decorators like this to enforce certain functionality. 

Should I open a new issue for this?
